Sp 1554

CeltonMcGrath opened the pull request:
Pipeline editor tab now detects changes in the pipeline canvas. This includes the pipeline editor menu items (refresh, save, add process etc.), sidebar navigator (to open other pipelines to view) and the close button.

Included is the detecting of changed rows in views which contained tables when opened in tabs.

oboreo commented on src/tactic/ui/tools/pipeline_wdg.py:
```diff
@@ -1402,8 +1401,11 @@ def get_display(my):
+            'type': 'click',
+            'cbjs_action': '''
+            '''
+        })
```
Whatis this empty behavior for?

CeltonMcGrath commented:
@oboreo Removed empty behaviour

oboreo commented on src/tactic/ui/container/tab_wdg.py:
```diff
@@ -844,27 +845,52 @@ def get_onload_js(my):
-        spt.tab.select(last);
+       ok();
     }
 }
```
I would remove changed_element and changed_type variables if they are not being used.. They are declared near the top as well. 
Can the message tell the user what kind of unsaved changes he has? like in tables or in UI

CeltonMcGrath replied:
The changed_element and changed_type variables are being used as they are parameters in the ok() function. However, I couldn't figure out how to pass them as parameters in spt.confirm

To implement the kind of changes, we could add an attribute to the div with "spt_has_changed" specifying the changes, but it depends how specific your the want the changes-description to be.

CeltonMcGrath commented:
Hi Boris, you were right about the Moodialog confirm argument passing. (Sorry for being lazy, will test proactively next time!) 

I've updated my usage in tab_wdg.py and added a little more doc in utility.js
